src/VisualStudio/Core/Def/Telemetry/TelemetryLogProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Telemetry;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.VisualStudio.LanguageServices.Telemetry
+{
+    /// <summary>
+    /// Provides access to an appropriate <see cref="ITelemetryLogProvider"/> for logging telemetry.
+    /// </summary>
+    internal sealed class TelemetryLogProvider : ITelemetryLogProvider, IDisposable
+    {
+        private readonly AggregatingTelemetryLogManager _aggregatingTelemetryLogManager;
+        private readonly VisualStudioTelemetryLogManager _visualStudioTelemetryLogManager;
+
+        private TelemetryLogProvider(TelemetrySession session, ILogger telemetryLogger)
+        {
+            _aggregatingTelemetryLogManager = new AggregatingTelemetryLogManager(session);
+            _visualStudioTelemetryLogManager = new VisualStudioTelemetryLogManager(telemetryLogger);
+        }
+
+        public static TelemetryLogProvider Create(TelemetrySession session, ILogger telemetryLogger)
+        {
+            var logProvider = new TelemetryLogProvider(session, telemetryLogger);
+
+            TelemetryLogging.SetLogProvider(logProvider);
+
+            return logProvider;
+        }
+
+        public void Dispose()
+        {
+            _aggregatingTelemetryLogManager.Dispose();
+        }
+
+        /// <summary>
+        /// Returns an <see cref="ITelemetryLog"/> for logging telemetry.
+        /// </summary>
+        public ITelemetryLog? GetLog(FunctionId functionId)
+        {
+            return _visualStudioTelemetryLogManager.GetLog(functionId);
+        }
+
+        /// <summary>
+        /// Returns an aggregating <see cref="ITelemetryLog"/> for logging telemetry.
+        /// </summary>
+        public ITelemetryLog? GetAggregatingLog(FunctionId functionId, double[]? bucketBoundaries = null)
```

Comment:
📝  This will be called through the interface, so the duplication of the default value is unnecessary
```suggestion
        public ITelemetryLog? GetAggregatingLog(FunctionId functionId, double[]? bucketBoundaries)
```

Reply:
Changed in commit 2.